packages/library-legacy-sham/src/__tests__/public-key-test.ts

```diff
@@ -0,0 +1,106 @@
+import { PublicKey } from '../public-key';
+
+const VALID_ADDRESS = 'TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb';
+const VALID_ADDRESS_BYTES = [
+    6, 221, 246, 225, 238, 117, 143, 222, 24, 66, 93, 188, 228, 108, 205, 218, 182, 26, 252, 77, 131, 185, 13, 39, 254,
+    189, 249, 40, 216, 161, 139, 252,
+];
+
+describe('PublicKeySham', () => {
+    it('throws when constructed using an invalid address', () => {
+        expect(() => {
+            new PublicKey('bad');
+        }).toThrow();
+    });
+    it.each([
+        'createProgramAddress',
+        'createProgramAddressSync',
+        'createWithSeed',
+        'decode',
+        'decodeUnchecked',
+        'isOnCurve',
+        'findProgramAddress',
+        'findProgramAddressSync',
```

Comment:
I expect people are likely gonna complain about these not being shamed.

Reply:
🙅🏼 _shame!_

Reply:
People won't use the sham. Libraries will use the sham (eg. `@metaplex/mpl-token-metadata`). So long as we support everything the _libraries_ need, we're done.